digdag-standards/src/main/java/io/digdag/standards/operator/aws/S3WaitOperatorFactory.java

```diff
@@ -168,27 +175,38 @@ public TaskResult run()
                 req.setSSECustomerKey(sseKey);
             }
 
-            ObjectMetadata objectMetadata = pollingWaiter(state, "EXISTS")
-                    .withPollInterval(POLL_INTERVAL)
-                    .withWaitMessage("Object '%s/%s' does not yet exist", bucket.get(), key.get())
-                    .await(pollState -> pollingRetryExecutor(pollState, "POLL")
-                            .retryUnless(AmazonServiceException.class, Aws::isDeterministicException)
-                            .run(s -> {
-                                try {
-                                    return Optional.of(s3Client.getObjectMetadata(req));
-                                }
-                                catch (AmazonS3Exception e) {
-                                    if (e.getStatusCode() == 404) {
-                                        return Optional.absent();
+            try {
+                ObjectMetadata objectMetadata = pollingWaiter(state, "EXISTS")
+                        .withPollInterval(POLL_INTERVAL)
+                        .withNextIntervalExpBase(NEXT_INTERVAL_EXP_BASE)
+                        .withTimeout(timeout)
+                        .withWaitMessage("Object '%s/%s' does not yet exist", bucket.get(), key.get())
+                        .await(pollState -> pollingRetryExecutor(pollState, "POLL")
+                                .retryUnless(AmazonServiceException.class, Aws::isDeterministicException)
+                                .run(s -> {
+                                    try {
+                                        return Optional.of(s3Client.getObjectMetadata(req));
+                                    } catch (AmazonS3Exception e) {
+                                        if (e.getStatusCode() == 404) {
+                                            return Optional.absent();
+                                        }
+                                        throw e;
                                     }
-                                    throw e;
-                                }
-                            }));
-
-            return TaskResult.defaultBuilder(request)
-                    .resetStoreParams(ImmutableList.of(ConfigKey.of("s3", "last_object")))
-                    .storeParams(storeParams(objectMetadata))
-                    .build();
+                                }));
+                return TaskResult.defaultBuilder(request)
+                        .resetStoreParams(ImmutableList.of(ConfigKey.of("s3", "last_object")))
+                        .storeParams(storeParams(objectMetadata))
+                        .build();
+            }
+            catch (PollingTimeoutException e) {
+                logger.debug(e.toString());
```

Comment:
Please keep in mind that this debug message is visible to users. At least we should add why a message is shown. Let's keep high quality of logging so that we don't get exhausted due to too many bad logs...

```suggestion
                logger.debug("s3_wait timed out: {} (making the task {})", (ignoreTimeoutError ? "continued" : "failed"), , e.toString());
```

Or if you think the message is really for development purpose only, then use trace level instead. Examples: https://github.com/treasure-data/digdag/search?q=logger.trace&unscoped_q=logger.trace